### PR TITLE
nfd-master: add validation of label names and values

### PR DESCRIPTION
**Pull request by marquiz**
Validate labels before trying to update the node. Makes us fail early nad prevent useless retries in case invalid labels are tried.

### CODE REVIEW

File: deployment/base/topologyupdater-daemonset/topologyupdater-daemonset.yaml
Comment (PiotrProkop) [inline deployment/base/topologyupdater-daemonset/topologyupdater-daemonset.yaml:1]: is this change required for this patch?

Comment (marquiz) [reply deployment/base/topologyupdater-daemonset/topologyupdater-daemonset.yaml:1]: No, definitely not! How on earth did this change and up in the PR 🤔 Fixed. Thanks for spotting this @PiotrProkop, looks like a bit hasty submission on my part...

**Comment by PiotrProkop** [comment]
I think we can experience -> https://github.com/kubernetes/test-infra/issues/29622
